numpy/ma/core.py

```diff
@@ -6535,9 +6530,7 @@ def compressed(x):
         Equivalent method.
 
     """
-    if not isinstance(x, MaskedArray):
-        x = asanyarray(x)
-    return x.compressed()
+    return asanyarray(x).compressed()
```

Comment:
In another PR is slightly better, but really this is making work, so whatever you feel like!